misc/output_test.py

```diff
@@ -371,6 +395,62 @@ def test_tool_inputs(self) -> None:
         )
 
 
+    def test_tool_compdb_targets(self) -> None:
+        self.maxDiff = 90000
+        plan = '''
+rule cat
+  command = cat $in $out
+build out1 : cat in1
+build out2 : cat in2 out1
+build out3 : cat out2 out1
+build out4 : cat in4
+'''
+
+        def test_expected_error(flags, expected):
+          actual = ''
+          try:
+            actual = run(plan, flags, print_err_output=False)
+          except subprocess.CalledProcessError as err:
+            actual = err.cooked_output
+          self.assertEqual(expected, actual)
```

Comment:
nit: I suggest you convert this into a standalone method, since `plan` is the only local variable that seems to be captured by this inner function, and that this functionaility could be reused elsewhere. Use a `_test_` prefix so that `unittest` doesn't think it is a test function, e.g.:

```
  def _test_expected_error(plan: str, flags : T.Union[str, None], expected: str):
     """Run Ninja with a given plan and flags, and verify its cooked output against an expected content.
     """
      ...
```
```

Reply:
i pretty much just copied & pasted this (though i changed the type of `flags` to `Optional[str]`); would you like me to add you as a co-author? 

(full disclosure: i have not touched python in years, so i've depended heavily on your suggestions here)